### ISSUE TITLE
Sieve connection failed on sieve://dovecot:4190/?tls=YES

### ISSUE THREAD

**Issue by Tsopic** [mailcow/mailcow-dockerized#2073]
**Describe the bug**
When using SOGO and trying to save mail user settings I encounter following error.
UI just says "Service temporarily unavailable"

```
sogo-mailcow_1       | Dec  6 15:10:20 f48986f13a56 sogod [59]: [ERROR] <0x0x557313156290[SOGoSieveManager]> Sieve connection failed on sieve://dovecot:4190/?tls=YES
```

**How to reproduce Reproduce**
Steps to reproduce the behavior please explain how to reproduce the bug:
1. change forward mail adressesses
2. try saving it

**Logs**
```
Dec  6 15:21:12 f48986f13a56 sogod [59]: [ERROR] <0x0x557312ed2b50[SOGoSieveManager]> Sieve connection failed on sieve://dovecot:4190/?tls=YES
Dec  6 15:21:12 f48986f13a56 sogod [59]: 27.79.228.10 "POST /SOGo/so/tsopic@example.ninja/Preferences/save HTTP/1.0" 503 46/3706 0.123 - - 0
```

**System**
 - OS:Ubuntu
 - Browser: Chrome
 - Other clients involved: letsencrypt-proxy-companyon

**Comment by andryyy** [comment]
Cannot confirm. Sieve is not broken, looks good in the demo too.

**Comment by Tsopic** [comment]
Ughh, been searching for hours. Need some glues on how to debug this issue.

**Comment by Tsopic** [comment]
Found it cheaper to just buy managed email host.

Closing this thread.

**Comment by badsmoke** [comment]
i currently have the same problem.

Mailcow is up to date

**Comment by andryyy** [comment]
You may want to join the Telegram channel and post some logs. :)

**Comment by badsmoke** [comment]
modify the sogo.conf 

from 
SOGoSieveServer = "sieve://dovecot:4190/?tls=YES"
to
SOGoSieveServer = "sieve://dovecot:4190/?tls=NO"

fixes the problem

**Comment by Ruff-G** [comment]
> modify the sogo.conf
> 
> from SOGoSieveServer = "sieve://dovecot:4190/?tls=YES" to SOGoSieveServer = "sieve://dovecot:4190/?tls=NO"
> 
> fixes the problem

Hello badsmoke,
i've tried the fix above, but nothing changed. I have the same problem like the issue-starter.
I press "Save" in the sogo-ui and after some seconds i get "Service temporarily unavailable".
Are there some solutions with this problem? It is a fresh and clean install from the master branch. I installed it yesterday and i'm looking for a solution for hours and found nothing.
In my logs of dovecot and sogo i can see following entries.

`sogo-mailcow_1       | Nov  3 14:24:19 307c5102c940 sogod [57]: <0x0x55b50b7d7b00[SOGoSieveManager]> failure. Attempting with a renewed password (no authname supported)`
`sogo-mailcow_1       | Nov  3 14:24:25 307c5102c940 sogod [57]: <0x0x55b50b7d7b00[SOGoSieveManager]> Could not login 'admin@exampe.com' on Sieve server: <0x0x55b50b903180[NGSieveClient]: socket=<NGActiveSocket[0x0x55b50b874c30]: mode=rw address=<0x0x55b50b86e4d0[NGInternetSocketAddress]: host=307c5102c940 port=49190> connectedTo=<0x0x55b50b85dc00[NGInternetSocketAddress]: host=dovecot port=4190>>>: {RawResponse = "{ok = 0; reason = \"Authentication failed.\"; }"; result = 0; }`
`sogo-mailcow_1       | Nov  3 14:24:25 307c5102c940 sogod [57]: 88.xxx.xxx.xxx "POST /SOGo/so/admin@example.com/Preferences/save HTTP/1.0" 503 46/3362 8.492 - - 2M - 13`

`dovecot-mailcow_1    | Nov  3 14:24:03 257162d85c43 dovecot: lmtp(896): Connect from 172.22.1.7`
`dovecot-mailcow_1    | Nov  3 14:24:03 257162d85c43 dovecot: lmtp(896): Disconnect from 172.22.1.7: Logged out (state=MAIL FROM)`
`dovecot-mailcow_1    | Nov  3 14:24:03 257162d85c43 dovecot: imap-login: Disconnected: Aborted login by logging out (no auth attempts in 0 secs): user=<>, rip=172.22.1.7, lip=172.22.1.250, TLS, TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)`
`dovecot-mailcow_1    | Nov  3 14:24:03 257162d85c43 dovecot: imap-login: Disconnected: Aborted login by logging out (no auth attempts in 0 secs): user=<>, rip=172.22.1.7, lip=172.22.1.250`
`dovecot-mailcow_1    | Nov  3 14:24:03 257162d85c43 dovecot: managesieve-login: Disconnected: Connection closed (no auth attempts in 0 secs): user=<>, rip=172.22.1.7, lip=172.22.1.250`
`dovecot-mailcow_1    | Nov  3 14:24:25 257162d85c43 dovecot: managesieve-login: Disconnected: Connection closed (auth failed, 2 attempts in 8 secs): user=<admin@example.com>, method=PLAIN, rip=172.22.1.248, lip=172.22.1.250`
`dovecot-mailcow_1    | Nov  3 14:24:31 257162d85c43 dovecot: imap-login: Login: user=<admin@example.com>, method=PLAIN, rip=172.22.1.248, lip=172.22.1.250, mpid=1069`
`dovecot-mailcow_1    | Nov  3 14:24:52 257162d85c43 dovecot: lmtp(896): Connect from 172.22.1.7`
`dovecot-mailcow_1    | Nov  3 14:24:52 257162d85c43 dovecot: lmtp(896): Disconnect from 172.22.1.7: Logged out (state=MAIL FROM)`
`dovecot-mailcow_1    | Nov  3 14:24:52 257162d85c43 dovecot: imap-login: Disconnected: Aborted login by logging out (no auth attempts in 0 secs): user=<>, rip=172.22.1.7, lip=172.22.1.250, TLS, TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)`
`dovecot-mailcow_1    | Nov  3 14:24:52 257162d85c43 dovecot: imap-login: Disconnected: Aborted login by logging out (no auth attempts in 0 secs): user=<>, rip=172.22.1.7, lip=172.22.1.250`
`dovecot-mailcow_1    | Nov  3 14:24:52 257162d85c43 dovecot: managesieve-login: Disconnected: Connection closed (no auth attempts in 0 secs): user=<>, rip=172.22.1.7, lip=172.22.1.250`

**Comment by fluxens** [comment]
@Ruff-G I had the same issue with a fresh install. The user did not have access to sieve in the allowed protocols in the mailbox settings. Giving access solved the issue.

<img width="564" alt="image" src="https://user-images.githubusercontent.com/22811368/140418887-1234711e-c1f9-4d9e-915d-a802771075f7.png">

**Comment by Ruff-G** [comment]
@fluxens
yes that it is. After several installations and looking up the problem, this was the solution. Big thanks. 👍

**Comment by strarsis** [comment]
It appears that a [regression happened](https://github.com/mailcow/mailcow-dockerized/issues/4398), as this issue happens again and I had to apply the manual workaround solution to fix it 
(https://github.com/mailcow/mailcow-dockerized/issues/2073#issuecomment-961415055).
How did this happen? Are there default settings that were changed during a mailcow-dockerized update in the past?
Can the default settings for new mailboxes be set somewhere? I want all newly created mailboxes to also have `Sieve` access.

**Comment by curdbecker** [comment]
I just ran into this problem and for me it appeared only for mailboxes that I created while their parent domain was a relay domain. For all other accounts that I created after, it worked. Maybe that's related to the cause of the issue. However, I can't reproduce it... so who knows. But the issue exists in fact and looks exactly like described. Also enabling Sieve access for the affected mailboxes manually fixes the issue.
